packages/core/redux/wallets/actions.js

```diff
@@ -78,73 +79,92 @@ export const initWallets = () => (dispatch) => {
 const initWalletsFromKeys = () => async (dispatch, getState) => {
   const state = getState()
   const account = getPersistAccount(state)
+  const addressCache = { ...getAddressCache(state) }
+
+  const signerSelectors = {
+    [BLOCKCHAIN_ETHEREUM]: getEthereumSigner,
+    [BLOCKCHAIN_BITCOIN]: getBitcoinSigner,
+    [BLOCKCHAIN_BITCOIN_CASH]: getBitcoinCashSigner,
+    [BLOCKCHAIN_DASH]: getDashSigner,
+    [BLOCKCHAIN_LITECOIN]: getLitecoinSigner,
+    [BLOCKCHAIN_NEM]: getNemSigner,
+    [BLOCKCHAIN_WAVES]: getWavesSigner,
+  }
+
+  Object.entries(signerSelectors).forEach(([ blockchain, signerSelector ]) => {
+    let address = addressCache[blockchain]
+    if(!address) {
+      const signer = signerSelector(state)
+      if (signer) {
+        address = signer.getAddress(accountPath)
```

Comment:
We need specify correct accountPath for each blockchain. E.g. derivation path for BTC is different in mainnet/testnet

Also, accountPath decalred only at line 109 below.